src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -61,13 +61,14 @@
   private final Set<ClassName> importableTypes = new LinkedHashSet<>();
   private boolean trailingNewline;
 
-  public CodeWriter(Appendable out) {
-    this(out, ImmutableMap.<ClassName, String>of());
+  public CodeWriter(Appendable out, String indent) {
+    this(out, ImmutableMap.<ClassName, String>of(), indent);
   }
 
-  public CodeWriter(Appendable out, ImmutableMap<ClassName, String> importedTypes) {
+  public CodeWriter(Appendable out, ImmutableMap<ClassName, String> importedTypes, String indent) {
```

Comment:
nit: reorder parameters 3 and 2 ? When there's overloads that make a parameter optional, that parameter should be last.